ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -156,11 +144,17 @@ private void verifyTableIdHasNotChanged(CompactionInfo ci, Table originalTable)
   public void init(AtomicBoolean stop) throws Exception {
     super.init(stop);
 
-    StringBuilder name = new StringBuilder(hostname());
+    StringBuilder name = new StringBuilder(ServerUtils.hostname());
     name.append("-");
     name.append(getId());
     this.workerName = name.toString();
     setName(name.toString());
+    this.runtimeVersion = getRuntimeVersion();
+  }
+
+  @VisibleForTesting
+  protected String getRuntimeVersion() {
```

Comment:
get runtime version could be moved to common place like CompactorThread with `class` as param.

Reply:
moved the runtimeVersion and host setup to CompactorThread